dask/array/rechunk.py

```diff
@@ -548,9 +569,9 @@ def plan_rechunk(
 
     if ndim <= 1 or not all(new_chunks) or any(has_nans):
         # Trivial array / unknown dim => no need / ability for an intermediate
-        return steps + [new_chunks]
+        return [new_chunks]
```

Comment:
I'm hesitant to inline `has_nans` since it would turn this into a three-line if statements that arguable harder to read. I've taken care of the other suggestions though.